jsonargparse/_actions.py

```diff
@@ -699,6 +699,25 @@ def get_subcommands(
             if subcommand is None and not (fail_no_subcommand and action._required):  # type: ignore
                 return None, None
             if action._required and subcommand not in action._name_parser_map:  # type: ignore
+                if dest == "subcommand":
+                    # If this action is a subcommand action and no subcommand is provided,
+                    # present the user with a friendly error message to remind them of
+                    # the available subcommands and to select one.
+                    available_subcommands = list(action._name_parser_map.keys())
+                    if len(available_subcommands) <= 10:
+                        candidate_subcommands_str = "{" + ",".join(available_subcommands) + "}"
+                    else:
+                        candidate_subcommands_str = (
+                            "{"
+                            + ",".join(available_subcommands[:5])
+                            + ", ..., "
+                            + ",".join(available_subcommands[-5:])
+                            + "}"
+                        )
+                    raise KeyError(
+                        f'expected "subcommand" to be one of {candidate_subcommands_str}, but it was not provided.'
```

Comment:
Don't hard code `"subcommand"`. You can see [add_subcommands](https://jsonargparse.readthedocs.io/en/v4.24.0/#jsonargparse.ArgumentParser.add_subcommands) in the docs, that subcommands can be added with any `dest` desired.

Reply:
Revised, please check